### PR TITLE
First pass

### PR DESCRIPTION
**Pull request by ryanthemanuel**
**Tech Design**

The initial purpose of the terra-toolkit will be to facilitate running functional tests in a headless browser with optional screenshots. In order to accomplish this, we will be spinning up the webpack dev server with a given test configuration, then set up a selenium web driver to interact with the running server via a headless browser (phantomjs). This set up will be integrated with jasmine so that if desired, screenshots can be taken automatically as well as referenced inside of a set of results that can (in theory) be used for visual regression testing later.

**API**

This toolkit will export two functions: testHelper and testRunner.

testHelper takes in a webpack configuration object that specifies the routes that will be needed for testing. testHelper's job is to be run once at the beginning of the test run. It will set up the jasmine test reporter to be jasmine-spec-reporter and then spin up a webpack dev server using the configuration that was passed in. This server will be running for the duration of the test run and then shut down at the end.

testRunner takes in an array of functions that will execute the actual test suite and an optional set of window sizes. If no set of window sizes is passed in, a default set of window sizes will be used. The window sizes will be iterated over, a selenium web driver will be created for that window size, and each function in the array of test functions will be executed. Thus, each test will be executed at each window size.

@cerner/terra

### CODE REVIEW

File: src/test-result.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import sha1 from 'sha1';
```
Comment (ryanthemanuel) [inline src/test-result.js]: Note that a lot of test-result.js and test-results.js is modeled after a corresponding rails development engine that has been used internally at Cerner to manage Capybara functional tests, their screenshots, and screenshot comparisons.  The idea is that this project will evolve over time to match the functionality in that project where it makes sense.

**Comment by jpruetting** [comment]
Looks good to me. I don't have a lot of opinion/expertise in this area, but I also don't think we need to be too picky with a first pass at this kind of work since it can only help and isn't going into use by consumers.

**Comment by ryanthemanuel** [comment]
@jpruetting agreed, I think since this is a tool rather than a client facing component it doesn't require as much scrutiny.  @mhemesath do you (or someone from your team) care to take a look?  I'd like to get this merged in when I can.

**Comment by mhemesath** [comment]
+1

**Pull request merged**